### PR TITLE
Vehicle Gyromats

### PR DESCRIPTION
**Pull request by Wunarg**
The following vehicles now work with gyromats: (Square root of effect)
Bomber
ArmoredBomber
ArmoredCar
Car
Catapult
SteamTank
Ballista

This does allow the car and the armored car to sort of fly at 2 to 3 gyromats due to how base kag handles vehicles, but considering the difficulty of obtaining gyromat and the durability of cars this seems fine, additionally this flying is quite hard to use

### CODE REVIEW

**Comment by Wunarg** [comment]
Additional Note: Only affects horizontal speed of ballons and ballons will drop them when you press the bomb release button similar to any item put into its inventory so its somewhat hard to use for bombing runs

**Comment by Wunarg** [comment]
Note: Other vehicles firing speed is not affected this is just the catapult since it seems the most mechanical

File: TFlippy_TerritoryControl_Core_Dev/Entities/Vehicles/BomberCommon.as [hunk 1]
```diff
@@ -318,6 +318,12 @@ void BomberHandling(CBlob@ this,VehicleInfo@ v)
 				force.x*=1.1f;
 			}
 		}
+
+		if(this.exists("gyromat_acceleration"))
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Core_Dev/Entities/Vehicles/BomberCommon.as]: space between if (

File: TFlippy_TerritoryControl_Core_Dev/Entities/Vehicles/VehicleCommon.as [hunk 2]
```diff
@@ -560,6 +560,12 @@ void Vehicle_StandardControls(CBlob@ this, VehicleInfo@ v)
 							moveForce *= 1.5f;
 						}
 
+						// add gyromat effect
+						if(this.exists("gyromat_acceleration"))
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Core_Dev/Entities/Vehicles/VehicleCommon.as]: and here

File: TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Catapult/Catapult.as [hunk 3]
```diff
@@ -86,7 +86,14 @@ void onTick(CBlob@ this)
 
 		if (v.cooldown_time > 0)
 		{
-			v.cooldown_time--;
+			if(this.exists("gyromat_acceleration"))
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Catapult/Catapult.as]: space

File: TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Catapult/Catapult.as [hunk 4]
```diff
@@ -86,7 +86,14 @@ void onTick(CBlob@ this)
...
+			{
+				v.cooldown_time = Maths::Max(0,v.cooldown_time-this.get_f32("gyromat_acceleration"));
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Catapult/Catapult.as]: space after comma

File: TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Catapult/Catapult.as [hunk 5]
```diff
@@ -151,17 +158,27 @@ bool Vehicle_canFire(CBlob@ this, VehicleInfo@ v, bool isActionPressed, bool was
...
 	{
-
-		if (charge < v.max_charge_time && isActionPressed)
+		
+		if (isActionPressed && charge < v.max_charge_time)
 		{
-			charge++;
-			v.charge = charge;
+			f32 chargeSpeed = 1;
+			if(this.exists("gyromat_acceleration"))
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Catapult/Catapult.as]: spacee

File: TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Catapult/Catapult.as [hunk 6]
```diff
@@ -219,6 +236,11 @@ void Vehicle_onFire(CBlob@ this, VehicleInfo@ v, CBlob@ bullet, const u8 _charge
 		vel += (Vec2f((_r.NextFloat() - 0.5f) * 128, (_r.NextFloat() - 0.5f) * 128) * 0.01f);
 		vel.RotateBy(angle);
 
+		if(this.exists("gyromat_acceleration"))
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Catapult/Catapult.as]: .

File: TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Catapult/Catapult.as [hunk 7]
```diff
@@ -230,6 +252,10 @@ void Vehicle_onFire(CBlob@ this, VehicleInfo@ v, CBlob@ bullet, const u8 _charge
 	// we override the default time because we want to base it on charge
 	int delay = 30 + (charge / (250 / 30));
 	v.fire_delay = delay;
+	if(this.exists("gyromat_acceleration"))
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Catapult/Catapult.as]: .

File: TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Hoverbike/Hoverbike.as [hunk 8]
```diff
@@ -75,6 +75,11 @@ void onTick(CBlob@ this)
 			
 			Vec2f vel = Vec2f(h, v);
 			
+			if(this.exists("gyromat_acceleration"))
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Weapons_Dev/Entities/Vehicles/Hoverbike/Hoverbike.as]: .

**Review by Vam-Jam — CHANGES_REQUESTED** [review]

**Comment by Wunarg** [comment]
I fixed the spaces

**Review by Vam-Jam — APPROVED** [review]

**Pull request merged**